Fix measurement discovery OOM - batch aggregation saves

## Problem

The `run-measurement-discovery` Makefile target is being killed with error 137 (OOM) when saving aggregation results to MongoDB.

**Error observed**:
```
[19:26:37] Found 63,999,329 (harmonized_name, content) pairs with ≥1 biosamples (ALL harmonized_names)
[19:26:37] Saving aggregation results to content_pairs_aggregated collection...
Killed
make: *** [Makefiles/measurement_discovery.Makefile:203: run-measurement-discovery] Error 137
```

**Root cause**: The script attempts to insert ~64 million documents at once using `insert_many()` on line 148 of `measurement_discovery_efficient.py`, loading all documents into memory before insertion.

## Solution

Implement batched inserts for aggregation results (same pattern as quantulum3 processing):
- Insert in batches of 10,000 documents
- Progress reporting every 1M documents
- Matches existing pattern in same file (lines 244-248)

This follows the same fix pattern used in issue #250 for `$addToSet` memory limits.

## Related Issues

- #250: Fix harmonized_name_usage_stats memory limit issues
- Memory management pattern established in recent commits (bee72dd, e92d7ca)

## Test Plan

- [ ] Run `make -f Makefiles/measurement_discovery.Makefile run-measurement-discovery`
- [ ] Verify aggregation save completes without OOM
- [ ] Verify 64M documents saved to `content_pairs_aggregated` collection
- [ ] Verify quantulum3 processing continues successfully

--- a/external_metadata_awareness/measurement_discovery_efficient.py
+++ b/external_metadata_awareness/measurement_discovery_efficient.py
@@ -136,17 +136,34 @@ def main(mongo_uri, min_count, progress_every, batch_size, limit, save_aggregati
         agg_collection = db['content_pairs_aggregated']
         print(f"[{time.strftime('%H:%M:%S')}] Saving aggregation results to content_pairs_aggregated collection...")
 
-        # Clear existing data and insert new results
+        # Clear existing data
         agg_collection.drop()
         if results:
-            # Add timestamp to each document
-            timestamped_results = []
-            for result in results:
-                result['aggregated_at'] = time.time()
-                timestamped_results.append(result)
+            # Add timestamp and save in batches to avoid memory issues (Issue #250)
+            batch = []
+            batch_size = 10000  # Insert 10k documents at a time
+            saved_count = 0
 
-            agg_collection.insert_many(timestamped_results)
-            print(f"[{time.strftime('%H:%M:%S')}] Saved {len(results):,} aggregation results")
+            for i, result in enumerate(results):
+                result['aggregated_at'] = time.time()
+                batch.append(result)
+
+                # Save batch when it reaches batch_size
+                if len(batch) >= batch_size:
+                    agg_collection.insert_many(batch)
+                    saved_count += len(batch)
+                    batch = []
+
+                    # Print progress every 1M documents
+                    if saved_count % 1000000 == 0:
+                        print(f"[{time.strftime('%H:%M:%S')}] Saved {saved_count:,}/{len(results):,} aggregation results...")
+
+            # Save final batch
+            if batch:
+                agg_collection.insert_many(batch)
+                saved_count += len(batch)
+
+            print(f"[{time.strftime('%H:%M:%S')}] Saved {saved_count:,} aggregation results")
         else:
             print(f"[{time.strftime('%H:%M:%S')}] No aggregation results to save")
 

